tests/test_config.py

```diff
@@ -0,0 +1,52 @@
+#! /usr/bin/env python
+# vim: expandtab shiftwidth=4 softtabstop=4 tabstop=17 filetype=python :
+import argparse
+import os
+import sys
+import unittest
+
+from caramel import config
+from tests import fixtures
+
+
+class TestConfig(unittest.TestCase):
+    def test_inifile_argument(self):
+        my_ini = "/a/path/to/my_ini_file.ini"
+
+        parser = argparse.ArgumentParser()
+        config.add_inifile_argument(parser)
+        args = parser.parse_args([my_ini])
+
+        self.assertEqual(args.inifile, my_ini)
+        return True
+
+    def test_inifile_env(self):
+        my_ini = "/a/path/to/my_ini_file.ini"
+
+        os.environ["CARAMEL_INI"] = my_ini
```

Comment:
This permanently changes the environment without restoring it after the test is done, thus if you run the tests in arbitary order, this change will make the  other tests behave differently.

The solution to that is usually to make  a function liek "add_inifile_argument"  take an incoming environment dict as it's argument, with the default being "os.environ" that can be overridden for the tests.

Reply:
My understanding was that changes made through os on the environment was not persistent? But I will defer to your judgement.

Reply:
Persistent is a different thing.  It may not persist after you exit the process, but it's certainly a global state for the current process runtime, and other tests that run later will definitely see changes.

A simple demonstration:

```python
#!/bin/env python3
import os
import unittest

class TestOne(unittest.TestCase):
    def test_one(self):
        assert "HUMBUG" not in os.environ
        os.environ["HUMBUG"] = "Bah. Humbug"

    def test_two(self):
        assert "HUMBUG" not in os.environ
        os.environ["HUMBUG"] = "Bah. Humbug"

    def test_three(self):
        assert "HUMBUG" not in os.environ
        os.environ["HUMBUG"] = "Bah. Humbug"


if __name__ == "__main__":
    unittest.main()
```